doc/template_definition.md

```diff
@@ -81,8 +124,79 @@ method.getBody().insertBegin(injectedCode);
 
 ```
 
-Kinds of templating
--------------------
+Definition of templates using `PatternBuilder`
```

Comment:
Patterns are more powerful than Template, so why don't you present them first?

Reply:
I suggest to refactor documentation later. First your understanding, then refactoring of code and then refactoring of documentation ... and note, I am mainly developer. I am not so good in writing good documentation - in communication.